# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webaudio/MediaBufferDecoder.cpp
# Commit: c0666de7deba
# Full Hash: c0666de7deba9de87fec896134ee86d9bc5b8ca7
# Author: az <azebrowski@mozilla.com>
# Date: 2024-11-09 09:32:30
# Regressor Bug: 1905287
# File Overlap Count: 1
# Description:
#   Bug 1905287 - Assert AudioInfo has at least as many channels as AudioData when a MediaDecodeTask finishes decoding to ensure sufficient buffer size is allocated. r=karlt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D228383
# ==============================================================================

diff -r 22a30eccffde -r c0666de7deba dom/media/webaudio/MediaBufferDecoder.cpp
--- a/dom/media/webaudio/MediaBufferDecoder.cpp	Fri Nov 08 15:35:37 2024 +0000
+++ b/dom/media/webaudio/MediaBufferDecoder.cpp	Fri Nov 08 15:42:05 2024 +0000
@@ -531,6 +531,11 @@
     const AudioDataValue* bufferData =
         static_cast<AudioDataValue*>(audioData->mAudioBuffer->Data());
 
+    // Channel count check for 1905287
+    MOZ_DIAGNOSTIC_ASSERT(audioData->mChannels <= channelCount,
+                          "MediaDecodeTask: "
+                          "AudioData has more channels than AudioInfo!");
+
     if (sampleRate != destSampleRate) {
       const uint32_t maxOutSamples = resampledFrames - writeIndex;
 
